### PR TITLE
Update Travis CI.

### PR DESCRIPTION
**Pull request by kevina**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,16 +5,16 @@ os:
...
 install: true
 
 before_install:
   - make deps
 
 script:
-  - go vet
-  - go test -v -race -coverprofile=coverage.txt -covermode=atomic .
+  - go vet ./...
+  - go test -v -race -coverprofile=coverage.txt -covermode=atomic ./...
```
Comment (Kubuxu) [inline .travis.yml]: You can replace that with:
```
go list -f '{{if (len .TestGoFiles)}}{{.ImportPath}}{{end}}' ./... | grep -v /vendor/ | xargs -L1 -i bash -c 'go test -v -race -coverprofile="coverage.$(echo "$1" | md5sum | head -c 16).txt" -covermode=atomic "$1"' "" {}
```

It got a bit complex but works, I just tested.

Comment (kevina) [reply .travis.yml]: What is this doing?

Comment (Kubuxu) [reply .travis.yml]: it lists packages with tests and then runs `go test -cover -coverprofile="$(path specific name)" $package` working around the fact that you can't run coverage with `./...`.

**Comment by Stebalien** [comment]
So... please don't throw a desk at me but... it may be worth it to just put an updatable script online somewhere and curl bash it. Something like:

```bash
#!/bin/bash
set -ex

# Get GX
go get github.com/whyrusleeping/gx
go get github.com/whyrusleeping/gx-go

# Install deps
gx --verbose install --global

# Rewrite
gx-go rewrite

# Vet
go vet ./...

# Test
if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
    go list -f '{{if (len .TestGoFiles)}}{{.ImportPath}}{{end}}' ./... \
        | grep -v /vendor/ \
        | while read pkg; do 
        profile="coverage.$(echo "$pkg" | md5sum | head -c 16).txt"
        go test -v -race -coverprofile="$profile" -covermode=atomic "$pkg"
    done
    # Doesn't count as a failure.
    bash <(curl -s https://codecov.io/bash) || true
else
    go test -v -race ./...
fi
```

And, for `.travis.yaml`, use:

```
language: go
os:
  - linux
  - osx

script:
  - bash <(curl -s https://raw.githubusercontent.com/ipfs/infrastructure/master/ci.sh)
```

That way, we can avoid having to manually go through and update everything.

**Comment by kevina** [comment]
@Stebalien I am not against that idea.  @whyrusleeping @Kubuxu thoughts?

**Comment by kevina** [comment]
I can set this up if we decide to go with it, but I will start with putting the script in a gist until I work out something that works for all the repos, then we can decide where to put it and I will semi-automatically update all the P.R.

Note, though that some of repo. such as https://github.com/libp2p/go-libp2p-crypt use a makefile.  The Makefile likely ends up doing the same thing, but its not a good idea to have the same code in two places as it is too easy to update one and forget about the other.

**Comment by Stebalien** [comment]
> Note, though that some of repo. such as https://github.com/libp2p/go-libp2p-crypt use a makefile, said makefile likely ends up doing the same thing, but its not a good idea to have the same code in two places as it is too easy to update one and forget about the other.

Yeah, lots of repos have makefiles that do the whole fetch gx dance. We could just leave that there call `make deps` in each repo (in the `before_script`) as we do now. I was trying to consolidate everything but yeah, that might make sense.

**Comment by whyrusleeping** [comment]
I like @Stebalien's idea. I'm fine moving forward with that.

**Comment by kevina** [comment]
@Stebalien @whyrusleeping should I add a very basic `Makefile` to each of the repos.  Thinking about adding a `deps` and `build` target at least.  Then the build script above will focus on just running the tests and coverage report.

Per our IRC discussion the common script will not enable `-race` in the tests by default due to speed concerns (especially on mac).  If it was in the existing `.travis.yml` file then it will be enabled.

**Comment by whyrusleeping** [comment]
@kevina that SGTM

**Comment by Stebalien** [comment]
SGTM as well. However, I'd make enabling `-race` an environment variable (easier to toggle on travis, can even be toggled through the web ui).

**Comment by Kubuxu** [comment]
@kevina that is great idea. I would put the script under `travis/ci.sh`, I am not sure about the repo but `ipfs/infrastructure` sounds about right.

@Stebalien thanks for cleaning up the ugliness of my script.

**Comment by kevina** [comment]
@Stebalien @whyrusleeping @Kubuxu for now the script is in this gist:
https://gist.github.com/kevina/0b81bb07379138c37b65f785a88b13c0

I will move it into a better place once I am happy with it and it works on all libp2p projects.

**Comment by kevina** [comment]
Per the irc discussion on \#ipfs-dev the scripts new home is https://github.com/ipfs/ci-helpers/blob/master/travis-ci/run-standard-tests.sh

**Review by Stebalien — APPROVED** [review]

**Pull request merged**